### PR TITLE
Support recent kernel versions

### PR DESCRIPTION
**Pull request by WayneLin1992**
Problem: Some error infos are from the gcc
Cause: The kernel is from 4.15 to 5.8, Some API was changed or missed.
Solution: Check the git from kernel, and refer vivid which is device driver in kernel.

### CODE REVIEW

**Review by jserv — CHANGES_REQUESTED** [review]
Revise the subject of pull request and/or git commits. You can say, "adapt newer kernel APIs." In addition, we have to ensure the support from kernel v4.15 (shipped with Ubuntu Linux 18.04), v5.4 (shipped with Ubuntu Linux 20.04), to recent versions (such as v5.8+).

File: Makefile [hunk 1]
```diff
@@ -1,15 +1,29 @@
-target = vcam
-vcam-objs = module.o control.o device.o videobuf.o fb.o
-obj-m = $(target).o
-
+KERNEL_GT_5_4 = $(shell expr `uname -r` \> "5.4.0-0-generic")
+KERNEL_GT_4_15 = $(shell expr `uname -r` \> "4.15.0-0-generic")
 CFLAGS_utils = -O2 -Wall -Wextra -pedantic -std=c99
 
-.PHONY: all
+ifeq (${KERNEL_GT_5_4},1)
```
Comment (jserv) [inline Makefile:5]: Don't do that. Change the C source files instead.

File: Makefile [hunk 2]
```diff
@@ -1,15 +1,29 @@
-target = vcam
-vcam-objs = module.o control.o device.o videobuf.o fb.o
-obj-m = $(target).o
-
+KERNEL_GT_5_4 = $(shell expr `uname -r` \> "5.4.0-0-generic")
```
Comment (jserv) [inline Makefile:1]: The build system should be consistent among different kernel versions.

File: Makefile [hunk 3]
```diff
@@ -1,15 +1,29 @@
...
+
+else ifeq (${KERNEL_GT_4_15},1)
+	target = vcam
+	vcam-objs = module.o control.o device.o videobuf.o fb.o
+	obj-m = $(target).o
 all: kmod vcam-util
 
+else
+
+all: kmod vcam-util
```
Comment (jserv) [inline Makefile:19]: Don't do that. If the kernel module fails to be built, it would altert.

**Comment by jserv** [comment]
Use `git rebase -i` to squash the commits.

**Comment by jserv** [comment]
After performing `git rebase`, do `git push --force`.

**Comment by jserv** [comment]
> Problem: Some error infos are from the gcc
> Cause: The kernel is from 4.15 to 5.8, Some API was changed or missed.
> Solution: Check the git from kernel, and refer vivid which is device driver in kernel.

Improve the above descriptions by editing.

**Review by jserv — CHANGES_REQUESTED** [review]
The current git commit messages look a bit redundant. You can concentrate on the verified kernel versions AND known issues if any.

**Comment by WayneLin1992** [comment]
I already compile on the ubuntu 21.04 which is on linux v5.11. vcam is working.

**Comment by jserv** [comment]
Thank @WayneLin1992 for contributing!

**Pull request merged**